Ew2d bvec

ludwig-ahrens opened the pull request:
Hey Robert,

der B-Vektor ist jetzt so weit, dass genau das gleiche raus kommt wie bei CONP.

Die Änderugen die ich gemacht habe um die Segmentation faults etc. zu beheben sind auch mit drin.

ludwig-ahrens committed:
Short range part of B-vector

ludwig-ahrens committed:
Debug memory problems

ludwig-ahrens committed:
Kspace of b_vec

ludwig-ahrens committed:
Fix pair_contribution of b_vec

ludwig-ahrens committed:
compute_coul_vector files

ludwig-ahrens committed:
compute_vector_corr() 3D slab correction

ludwig-ahrens committed:
Merge branch 'ew2d' into ew2d-bvec

robeme commented on src/USER-CONP/compute_coul_vector.cpp:
```diff
@@ -0,0 +1,433 @@
+  }
+}
+/* ---------------------------------------------------------------------- */
+void ComputeCoulVector::matrix_assignment() {
```
Um etwas konsistent zu sein, sollte das eigentlich eher vector_assignment sein und statt mpos vpos oder?

ludwig-ahrens committed:
Renaming from coul to conp and small stuff

compute_vector not compute_array in coul_vector
mpos has size nlocal instead of ngroup

robeme commented on src/USER-CONP/ewald_conp.cpp:
```diff
@@ -1634,6 +1636,76 @@ void EwaldConp::ew2dcorr_groups(int /*groupbit_A*/, int /*groupbit_B*/,
+      bi += 2 * ug[k] * (cos_kr * q_cos[k] + sin_kr * q_sin[k]);
+      // different sign than fix_conp for now
+    }
+    vector[imat[i]] += bi;
```
Ähnlich hier: imat -> ivec?

robeme commented on src/USER-CONP/ewald_conp.cpp:
```diff
@@ -1810,32 +1871,36 @@ void EwaldConp::compute_matrix(bigint *imat, double **matrix) {
             snx_all[kxj] * csy_all[kyj] + csx_all[kxj] * sny_all[kyj] * sign_ky;
 
-        cos_kxkykz_j =
+        double const cos_kxkykz_j =
```
Ist es OK in einem Loop jedesmal einen double const zu initialisieren?

ludwig-ahrens replied:
Es funktioniert jedenfalls. Ob das klug ist weiß ich nicht.

robeme replied:
Mh, OK, check nochmal, ob es so Sinn macht, oder ob man lieber einmal ausserhalb des Loops cos_kxkykz etc definiert und sie nachher nur überschreibt... Könnte u.U. schneller sein.

robeme commented on src/USER-CONP/ewald_conp.cpp:
```diff
@@ -1846,7 +1911,7 @@ void EwaldConp::compute_matrix(bigint *imat, double **matrix) {
     }
 
-    if ((i + 1) % 100 == 0) printf("(%d/%d) on %d\n", i + 1, nlocal, comm->me);
+    if ((i + 1) % 500 == 0) printf("(%d/%d) on %d\n", i + 1, nlocal, comm->me);
```
Das kann im Prinzip am Ende ganz raus... Ist erstmal nur für uns, damit wir sehen, wie weit er ist.

robeme reviewed (commented):
Lass uns generell alles in conp/vector, conp/matrix, etc... umbennen.

ludwig-ahrens commented:
> Lass uns generell alles in conp/vector, conp/matrix, etc... umbennen.

Hab ich gemacht.
Und ich habe die Größe von mpos von ngroup zu nlocal geändert.
